src/getSchema.ts

```diff
@@ -14,12 +15,18 @@ import type { CstNodeLocation } from 'chevrotain';
  * // ... make changes to schema object ...
  * const changedSource = printSchema(schema)
  * */
-export function getSchema(source: string): Schema {
+export function getSchema(
+  source: string,
+  configOverride?: PrismaAstConfig
```

Comment:
I'm not opposed to this approach, however, one of the objectives of the current design is that it re-uses the same parser for multiple getSchema() calls, to retain a good performance benchmark when called in bulk (e.g. if you have an entire directory of schema files)

I'd propose that, instead of passing in the config and creating the parser inside this method, you pass in the parser itself. That way prisma-lint can export its own `LintParser` object, as a singleton, and reuse it.

```ts
import { PrismaLexer } from './lexer';
import { parser as prismaAstParser } from './parser';
import { visitorClassFactory } from './visitor';
import type { CstNodeLocation } from 'chevrotain';
import getConfig, { PrismaAstConfig } from './getConfig';
import { PrismaParser } from './parser';

export function getSchema(
  source: string,
  options: { parser: typeof prismaAstParser } = { parser: prismaAstParser }
): Schema {
  const { parser } = options;
  const lexingResult = PrismaLexer.tokenize(source);
  parser.input = lexingResult.tokens;
  const cstNode = parser.schema();
  if (parser.errors.length > 0) throw parser.errors[0];
  const VisitorClass = visitorClassFactory(parser);
  const visitor = new VisitorClass(config);
  return visitor.visit(cstNode);
}
```

Reply:
Actually, the same could be said about the visitor. The visitor instance can be passed in as an option, too. Though your visitor factory will also need to be exported